### PR TITLE
Factored out WalkParams

### PR DESCRIPTION
**Pull request by Conorbro**
Factored out WalkParams into separate struct to be inlined in config.Module and ModuleConfig

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -64,31 +70,31 @@ func (c *Module) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err := CheckOverflow(c.XXX, "module"); err != nil {
 		return err
 	}
-	if c.Auth == nil {
-		c.Auth = &DefaultAuth
+	if c.WalkParams.Auth == nil {
```
Comment (brian-brazil) [inline config/config.go]: All these checks should be done in the Unmarshal of WalkParams

File: config/config.go [hunk 2]
```diff
@@ -64,31 +70,31 @@ func (c *Module) UnmarshalYAML(unmarshal func(interface{}) error) error {
...
+		c.WalkParams.Auth = &DefaultAuth
```
Comment (brian-brazil) [inline config/config.go]: Use the default logic we're using elsewhere.

File: config/config.go [hunk 3]
```diff
@@ -97,19 +103,19 @@ func (c *Module) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // configureSNMP sets the various version and auth settings.
 func (c Module) ConfigureSNMP(g *gosnmp.GoSNMP) {
```
Comment (brian-brazil) [inline config/config.go]: This can change to be a method of WalkParams

**Comment by Conorbro** [comment]
@brian-brazil

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**